### PR TITLE
[HUE-5420] Support alternative s3 endpoints.

### PR DESCRIPTION
**Pull request by ehiggs**
This is done by exposing the following configuration options:
  * proxy_address - address of the endpoint
  * proxy_port - port to use on the endpoint
  * is_secure - determine whether we're using https or http
  * calling_format - allow OrdinaryCallingFormat since not all endpoints have DNS
setup to support `<bucket-name>.domain/<file>`.

### CODE REVIEW

File: desktop/conf.dist/hue.ini [hunk 1]
```diff
@@ -1277,6 +1277,18 @@
       # AWS region to use
       ## region=us-east-1
 
+      # Endpoint overrides
```
Comment (romainr) [inline desktop/conf.dist/hue.ini]: Could you add the same in https://github.com/cloudera/hue/blob/master/desktop/conf/pseudo-distributed.ini.tmpl too? (those are almost the same ini files to keep in sync)

File: desktop/libs/aws/src/aws/conf.py [hunk 2]
```diff
@@ -91,6 +92,28 @@ def get_default_region():
         key='region',
         default='us-east-1',
         type=str
+      ),
+      PROXY_ADDRESS=Config(
+        help='Alternate address for endpoint.',
```
Comment (romainr) [inline desktop/libs/aws/src/aws/conf.py]: Could you use _() on all the help string to internationalize them? 
(and just noticed we forgot to do this in one above :)

**Comment by ehiggs** [comment]
> Sign Ccla

Done

> Copy into ...tmpl

Done!

> Use `_(...)` in help strings

Done. I also wrapped other help strings in the config as you said others had omitted them.

Thank you for taking a look at my contribution. I'm excited to help this project and had fun hacking on it.

**Comment by romainr** [comment]
Thanks for the contribution!

**Pull request merged**